diff --git a/fortran/src/CMakeLists.txt b/fortran/src/CMakeLists.txt
index 613b90b..e6876ee 100644
--- a/fortran/src/CMakeLists.txt
+++ b/fortran/src/CMakeLists.txt
@@ -127,17 +127,26 @@ if (BUILD_STATIC_LIBS)
 endif ()
 
 #-----------------------------------------------------------------------------
+set(H5MATCH_TYPES_BIN "$ENV{H5MATCH_TYPES_BIN}" CACHE STRING "binary containing pre-compiled H5match_types executable")
+if (H5MATCH_TYPES_BIN)
+    add_custom_target (H5match_types ALL
+        COMMAND ${H5MATCH_TYPES_BIN}
+        WORKING_DIRECTORY ${HDF5_F90_BINARY_DIR}
+        BYPRODUCTS ${HDF5_F90_BINARY_DIR}/H5f90i_gen.h ${HDF5_F90_BINARY_DIR}/H5fortran_types.F90
+        COMMENT "Running pre-compiled H5match_types binary"
+    )
+else()
 add_executable (H5match_types
     ${HDF5_F90_BINARY_DIR}/H5fort_type_defines.h
     ${HDF5_F90_SRC_SOURCE_DIR}/H5match_types.c
 )
 target_include_directories (H5match_types PRIVATE "${HDF5_SRC_BINARY_DIR};${HDF5_SRC_INCLUDE_DIRS};${HDF5_F90_BINARY_DIR};$<$<BOOL:${HDF5_ENABLE_PARALLEL}>:${MPI_C_INCLUDE_DIRS}>")
-
 add_custom_command (TARGET H5match_types POST_BUILD
     BYPRODUCTS ${HDF5_F90_BINARY_DIR}/H5f90i_gen.h ${HDF5_F90_BINARY_DIR}/H5fortran_types.F90
-    COMMAND ${CMAKE_CROSSCOMPILING_EMULATOR} $<TARGET_FILE:H5match_types>
+    COMMAND $<TARGET_FILE:H5match_types>
     WORKING_DIRECTORY ${HDF5_F90_BINARY_DIR}
 )
+endif()
 
 if (BUILD_STATIC_LIBS)
   add_custom_command (TARGET H5match_types POST_BUILD
