--- hdf5-1.10.2.orig/testpar/Makefile.in	2018-07-02 15:10:14.000000000 +0200
+++ hdf5-1.10.2/testpar/Makefile.in	2018-07-02 15:11:52.000000000 +0200
@@ -797,7 +797,7 @@

 # Test programs.  These are our main targets.
 #
-TEST_PROG_PARA = t_mpi t_bigio t_pflush1 t_pflush2 t_pread t_pshutdown t_prestart t_init_term t_filters_parallel
+TEST_PROG_PARA = t_mpi t_bigio t_pread t_pshutdown t_prestart t_init_term t_filters_parallel
 testphdf5_SOURCES = testphdf5.c t_dset.c t_file.c t_file_image.c t_mdset.c \
    t_ph5basic.c t_coll_chunk.c t_span_tree.c t_chunk_alloc.c t_filter_read.c \
    t_prop.c
