From b6be33ad68b84ba4e28bb25ae40028cc23c7b696 Mon Sep 17 00:00:00 2001
From: Min RK <benjaminrk@gmail.com>
Date: Wed, 14 Jun 2023 23:33:31 +0200
Subject: [PATCH 1/6] skip testphdf5

---
 testpar/Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/testpar/Makefile.am b/testpar/Makefile.am
index 4a8cb826f4..901c8258d6 100644
--- a/testpar/Makefile.am
+++ b/testpar/Makefile.am
@@ -33,7 +33,7 @@ check_SCRIPTS = $(TEST_SCRIPT_PARA)
 
 # Test programs.  These are our main targets.
 #
-TEST_PROG_PARA=t_mpi t_bigio testphdf5 t_cache t_cache_image t_pread t_pshutdown t_prestart t_init_term t_pmulti_dset t_select_io_dset t_shapesame t_filters_parallel t_2Gio t_vfd
+TEST_PROG_PARA=t_mpi t_bigio t_cache t_cache_image t_pread t_pshutdown t_prestart t_init_term t_pmulti_dset t_select_io_dset t_shapesame t_filters_parallel t_2Gio t_vfd
 
 if SUBFILING_VFD_CONDITIONAL
     TEST_PROG_PARA += t_subfiling_vfd
-- 
2.25.1

