diff --git a/fortran/src/Makefile.am b/fortran/src/Makefile.am
index 2fdf085f4b..ba7de839c1 100644
--- a/fortran/src/Makefile.am
+++ b/fortran/src/Makefile.am
@@ -121,13 +121,13 @@ BUILT_SOURCES = H5f90i_gen.h
 # header, then H5match_types which includes that header, then
 # it needs to run H5match_types.
 H5fortran_types.F90 H5f90i_gen.h: H5match_types$(EXEEXT)
-	$(RUNSERIAL) ./H5match_types$(EXEEXT)
+	$(RUNSERIAL) $(src_run)H5match_types$(EXEEXT)
 
 # H5_buildiface.F90 generates all the APIs that have a KIND type associated
 # with them.
 
 H5_gen.F90: H5_buildiface$(EXEEXT)
-	$(RUNSERIAL) ./H5_buildiface$(EXEEXT)
+	$(RUNSERIAL) $(src_run)H5_buildiface$(EXEEXT)
 
 # H5_buildiface.F90 is included in the distribution, and Automake knows
 # how to compile a fortran program given its sources.
diff --git a/hl/fortran/src/Makefile.am b/hl/fortran/src/Makefile.am
index 5834c9932d..9b5d1d9dad 100644
--- a/hl/fortran/src/Makefile.am
+++ b/hl/fortran/src/Makefile.am
@@ -95,7 +95,7 @@ noinst_PROGRAMS = H5HL_buildiface
 # with them.
 
 H5LTff_gen.F90: H5HL_buildiface$(EXEEXT)
-	$(RUNSERIAL) ./H5HL_buildiface$(EXEEXT)
++	$(RUNSERIAL) $(src_run)H5HL_buildiface$(EXEEXT)
 
 H5TBff_gen.F90: H5HL_buildiface$(EXEEXT)
 
