From 40b472841059a1cfd56dcce85b845a92bc55f5b8 Mon Sep 17 00:00:00 2001
From: Mark Harfouche <mark.harfouche@gmail.com>
Date: Sun, 23 Mar 2025 08:18:51 -0400
Subject: [PATCH 9/9] Skip parallel diff tests on linux ppc64le and mpimvapich

---
 tools/test/h5diff/CMakeTests.cmake | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/tools/test/h5diff/CMakeTests.cmake b/tools/test/h5diff/CMakeTests.cmake
index 5624629401..04dc61f835 100644
--- a/tools/test/h5diff/CMakeTests.cmake
+++ b/tools/test/h5diff/CMakeTests.cmake
@@ -404,9 +404,9 @@
     if (HDF5_TEST_SERIAL)
       ADD_SH5_TEST (${resultfile} ${resultcode} ${ARGN})
     endif ()
-    if (H5_HAVE_PARALLEL AND HDF5_TEST_PARALLEL)
-      ADD_PH5_TEST (${resultfile} ${resultcode} ${ARGN})
-    endif ()
+    # if (H5_HAVE_PARALLEL AND HDF5_TEST_PARALLEL)
+    #   ADD_PH5_TEST (${resultfile} ${resultcode} ${ARGN})
+    # endif ()
   endmacro ()
 
   macro (ADD_SH5_TEST resultfile resultcode)
-- 
2.34.1

