diff --git a/hl/fortran/src/Makefile.in b/hl/fortran/src/Makefile.in
index 66dfc10a56..197ec24d79 100644
--- a/hl/fortran/src/Makefile.in
+++ b/hl/fortran/src/Makefile.in
@@ -113,6 +113,14 @@ host_triplet = @host@
 # pass the -static flag to the library linker.
 @FORTRAN_SHARED_CONDITIONAL_FALSE@am__append_1 = -static
 noinst_PROGRAMS = H5HL_buildiface$(EXEEXT)
+
+cross_compiling = @cross_compiling@
+ifeq ($(cross_compiling),yes)
+  src_run = 
+else
+  src_run = ./
+endif
+
 TESTS =
 
 # log VFD currently has file space allocation bugs
@@ -1473,7 +1481,7 @@ uninstall-local:
 # with them.
 
 H5LTff_gen.F90: H5HL_buildiface$(EXEEXT)
-	$(RUNSERIAL) ./H5HL_buildiface$(EXEEXT)
+	$(RUNSERIAL) $(src_run)H5HL_buildiface$(EXEEXT)
 
 H5TBff_gen.F90: H5HL_buildiface$(EXEEXT)
 # Hardcode the dependencies of these files. There isn't a known way of
