--- a/testpar/Makefile.in	2023-05-09 22:00:29.838264540 -0400
+++ b/testpar/Makefile.in	2023-05-09 22:00:46.773990112 -0400
@@ -908,7 +908,7 @@
 
 # Test programs.  These are our main targets.
 #
-TEST_PROG_PARA = t_mpi t_bigio testphdf5 t_cache t_cache_image t_pread \
+TEST_PROG_PARA = t_mpi t_bigio t_cache t_cache_image t_pread \
 	t_pshutdown t_prestart t_init_term t_pmulti_dset \
 	t_select_io_dset t_shapesame t_filters_parallel t_2Gio t_vfd \
 	$(am__append_2)
