--- work/testpar/Makefile.in	2022-12-28 16:18:34.000000000 -0500
+++ b/testpar/Makefile.in	2023-01-15 08:28:46.515667050 -0500
@@ -904,7 +904,7 @@
 
 # Test programs.  These are our main targets.
 #
-TEST_PROG_PARA = t_mpi t_bigio testphdf5 t_cache t_cache_image t_pread \
+TEST_PROG_PARA = t_mpi t_bigio t_cache t_cache_image t_pread \
 	t_pshutdown t_prestart t_init_term t_pmulti_dset t_shapesame \
 	t_filters_parallel t_2Gio t_vfd $(am__append_2)
 testphdf5_SOURCES = testphdf5.c t_dset.c t_file.c t_file_image.c t_mdset.c \
