--- hdf5-1.12.0.orig/testpar/Makefile.in	2020-07-18 17:38:20.176820078 -0500
+++ hdf5-1.12.0/testpar/Makefile.in	2020-07-18 17:39:27.450792118 -0500
@@ -817,7 +817,7 @@
 
 # Test programs.  These are our main targets.
 #
-TEST_PROG_PARA = t_mpi t_bigio testphdf5 t_cache t_cache_image t_pread t_pshutdown t_prestart t_init_term t_shapesame t_filters_parallel t_2Gio
+TEST_PROG_PARA = t_mpi t_bigio t_pread t_pshutdown t_prestart t_init_term t_2Gio
 testphdf5_SOURCES = testphdf5.c t_dset.c t_file.c t_file_image.c t_mdset.c \
 	t_ph5basic.c t_coll_chunk.c t_span_tree.c t_chunk_alloc.c t_filter_read.c \
 	t_prop.c t_coll_md_read.c
