--- a/test/Makefile.in
+++ b/test/Makefile.in
@@ -1590,8 +1590,8 @@ check_SCRIPTS = $(TEST_SCRIPT)
 # As an exception, long-running tests should occur earlier in the list.
 # This gives them more time to run when tests are executing in parallel.
 TEST_PROG = testhdf5 \
-           cache cache_api cache_image cache_tagging lheap ohdr \
-           stab gheap evict_on_close farray earray btree2 fheap \
+           cache_api cache_tagging lheap ohdr \
+           stab gheap evict_on_close farray earray btree2 \
            accum hyperslab istore bittests dt_arith page_buffer \
            dtypes dsets chunk_info cmpd_dset cmpd_dtransform filter_fail extend direct_chunk \
            external efc objcopy objcopy_ref links unlink twriteorder big mtime \
